frontend/src/metabase-lib/lib/Dimension.js

```diff
@@ -752,7 +752,7 @@ export class FieldDimension extends Dimension {
     }
 
     // Add temporal dimensions
-    if (field.isDate()) {
+    if (field.isDate() && !this.isIntegerFieldId()) {
```

Comment:
```suggestion
    if (field.isDate() && this.isStringFieldName()) {
```

Not a big deal, but we could flip the logic to remove `!`.

Reply:
Also now that you mention that, I actually change in mind and would like to attempt the same fix as PR #15530, i.e. if this is a FieldIDDimension, the control flow shouldn't even reach this point.